src/main/ngapp/accio/src/app/editor/mask-controller.ts

```diff
@@ -1,27 +1,70 @@
-import { SetOperator } from './set-operator';
+import { MaskAction } from './mask-action';
+
+export enum Status {
+  STATUS_FAILURE = -1,
+  STATUS_SUCCESS = 0,
+}
+
+export enum Move {
+  BACK_ONE = -1,
+  FORWARD_ONE = 1,
+}
 
 export class MaskController {
-  private mask: Set<number> = new Set();
-  // Liable to change element datatype; design has not been determined...
-  history: Array<Set<number>> = [];
-  pPresent: number = 0;
+  private history: Array<MaskAction> = [];
+  // pPresent at an invalid index until an action is performed
+  private pPresent: number = -1;
 
-  constructor() { }
+  constructor(private mask: Set<number> = new Set()) {}
 
-  // Cannot access mask directly, so that a mask can never be accidentally
-  // altered without also changing its history.
-  getMask(): Set<number> {
-    // Change this implementation if undo/redo design is changed
+  public getMask(): Set<number> {
```

Comment:
Can you add a comment here to clarify that you're getting the "present" mask, or which version of the mask for this function.
Other than that, lgtm